src/pip/_internal/models/link.py

```diff
@@ -180,6 +181,16 @@ def show_url(self):
         # type: () -> Optional[str]
         return posixpath.basename(self._url.split('#', 1)[0].split('?', 1)[0])
 
+    @property
+    def is_file(self):
+        # type: () -> bool
+        return self.scheme == 'file'
+
+    @property
+    def is_existing_dir(self):
```

Comment:
I would make this a method rather than property since it's hitting the file system (calling `os.path.isdir()`).

Reply:
I'm scared of the implications of that. :) We could argue that `is_vcs` should be a method for the same reason.

Reply:
> We could argue that is_vcs should be a method

That'd be okay with me. I'd rather err on the side of being a method rather than a property..

Reply:
Incidentally, we should probably work to remove that nested import, anyways.

Reply:
Can we defer to a follow-up? Personally I'd rather keep the consistent API, but if the changes are in bulk I think it would be OK.

Reply:
Yea, let's do that in a follow up.

Reply:
Removing the nested import can of course be done later. I don't think we should tie making `is_existing_dir` a property with removing the nested import from `is_vcs` though. I think it's okay for some things to be properties and some things to be methods, depending on what they do.

Reply:
Addressed [here](https://github.com/pypa/pip/pull/7036/commits/892b58f2442270f1748f31b9dc9faeba5b14a79c#diff-f4abd4fac6833a14ebbea3016ce9a945R189).

Reply:
Done [here](https://github.com/pypa/pip/pull/7036/files#diff-f4abd4fac6833a14ebbea3016ce9a945R189).